File: plumbot-bukkit/src/main/java/me/regadpole/plumbot/internal/WhitelistHelper.java
package me.regadpole.plumbot.internal;

import me.regadpole.plumbot.PlumBot;
import me.regadpole.plumbot.config.Config;
import me.regadpole.plumbot.config.DataBase;
import me.regadpole.plumbot.internal.database.Database;
import me.regadpole.plumbot.internal.database.DatabaseManager;

import java.util.List;

public class WhitelistHelper {

    public static boolean checkCount(String qq){
        List<String> idList = DatabaseManager.getBind(qq, DataBase.type().toLowerCase(), PlumBot.getDatabase());
        int maxCount = Config.WhiteListMaxCount();
        if (idList.isEmpty()) return true;
        return idList.size() < maxCount;
    }

    public static boolean checkIDNotExist(String id){
        return DatabaseManager.getBindId(id, DataBase.type().toLowerCase(), PlumBot.getDatabase()) == 0L;
    }

    public static List<String> addAndGet(String id, String qq, String mode, Database db){
        DatabaseManager.addBind(id, qq, mode, db);
        return DatabaseManager.getBind(qq, mode, db);
    }

    public static List<String> removeAndGet(String qq, int num, String mode, Database db){
        DatabaseManager.removeBind(qq, num, mode, db);
        return DatabaseManager.getBind(qq, mode, db);
    }
}

File: common/src/main/kotlin/me/regadpole/plumbot/internal/WhitelistHelper.kt
package me.regadpole.plumbot.internal

import me.regadpole.plumbot.PlumBot


object WhitelistHelper {
    fun checkCount(user: String): Boolean {
        val idList = PlumBot.getDatabase().getBind(user).values
        val maxCount: Int = PlumBot.getConfig().getConfig().groups.forwarding.whitelist.maxCount
        if (idList.isEmpty()) return true
        return idList.size < maxCount
    }

    fun checkIDNotExist(name: String): Boolean {
        return PlumBot.getDatabase().getBindByName(name).isNullOrEmpty()
    }

    fun checkIDBelong(user: String, name: String): Boolean {
        return PlumBot.getDatabase().getBindByName(name) == user
    }

    fun addAndGet(user: String, name: String): MutableMap<Int, String> {
        PlumBot.getDatabase().addBind(user, name)
        return PlumBot.getDatabase().getBind(user)
    }

    fun removeAndGet(user: String, id: Int): MutableMap<Int, String> {
        PlumBot.getDatabase().removeBind(user, id)
        return PlumBot.getDatabase().getBind(user)
    }
}